add-on/src/popup/quick-upload.js

```diff
@@ -61,34 +59,28 @@ async function processFiles (state, emitter, files) {
     }
     const { ipfsCompanion } = await browser.runtime.getBackgroundPage()
     const uploadTab = await browser.tabs.getCurrent()
-    const { streams, totalSize } = files2streams(files)
-    progressHandler(0, totalSize, state, emitter)
+    const streams = files2streams(files)
     emitter.emit('render')
-    const wrapFlag = (state.wrapWithDirectory || streams.length > 1)
     const options = {
-      progress: (len) => progressHandler(len, totalSize, state, emitter),
-      wrapWithDirectory: wrapFlag,
-      pin: state.pinUpload
+      create: true,
+      parents: true
     }
-    let result
+    state.progress = `Uploading ${streams.length} files...`
```

Comment:
Note to self: it looks like we are losing import progress report, but it never worked correctly over HTTP API (https://github.com/ipfs/js-ipfs-http-client/issues/842), so its ok to drop it for now.